### PR TITLE
TMA-563: Provisioning brick doesn't verify status of project_id

### PR DESCRIPTION
**Pull request by vhtien**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/93/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/941/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/87/ : SUCCESS in 21s

**Comment by panjan** [comment]
ok to merge

File: lib/gooddata/lcm/actions/collect_clients.rb [hunk 1]
```diff
@@ -83,10 +85,22 @@ def collect_clients(params, segment_names = nil)
             segment_name = row[segment_id_column]
             GoodData.logger.debug("Segment name: #{segment_name}")
             if segment_names.nil? || segment_names.include?(segment_name)
+              client_id = row[client_id_column]
+              pid = row[project_id_column]
+
+              if pid
+                begin
+                  errors << "Project #{pid} of client #{client_id} is deleted." if client.projects(pid).deleted?
```
Comment (panjan) [inline lib/gooddata/lcm/actions/collect_clients.rb]: `client.projects(pid).deleted?` will make a call to the API for each client. Can we reuse `CollectClientProjects` from https://github.com/gooddata/gooddata-ruby/pull/986?

Comment (vhtien) [inline lib/gooddata/lcm/actions/collect_clients.rb]: nope, it's not! This project is not from any clients. This is an abandon project and user want to assign it to a new client in provisioning brick. So this is the right place to check it.

Comment (panjan) [inline lib/gooddata/lcm/actions/collect_clients.rb]: I see. Thanks for clarification,

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/43/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/948/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/94/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/855/ : SUCCESS in 45m 00s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/628/ : FAILURE in 1m 11s

**Comment by panjan** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/48/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/957/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/103/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/860/ : FAILURE in 45m 26s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/633/ : FAILURE in 48s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/103/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/861/ : SUCCESS in 46m 14s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/634/ : FAILURE in 54s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/49/ : SUCCESS in 49s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/961/ : SUCCESS in 49s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/107/ : SUCCESS in 22s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/862/ : SUCCESS in 44m 34s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/636/ : FAILURE in 2m 42s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/53/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/976/ : SUCCESS in 35s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/122/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/870/ : SUCCESS in 46m 09s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/644/ : SUCCESS in 19m 58s

**Pull request merged**